packages/plugin-types/src/resource_access_acceptor.ts

```diff
@@ -0,0 +1,30 @@
+import { Policy } from 'aws-cdk-lib/aws-iam';
+
+export type SsmEnvironmentEntry = {
+  /**
+   * The environment variable name where this SSM value will be placed at runtime
+   */
+  name: string;
+  /**
+   * The SSM parameter path that will be fetched at runtime
+   */
+  path: string;
+};
+
+export type ResourceAccessAcceptor = (
+  policy: Policy,
+  ssmEnvironmentEntries: SsmEnvironmentEntry[]
+) => void;
+
+export type ResourceAccessAcceptorFactory<
```

Comment:
This is a new type that our Constructs can implement in order for policies to be attached to roles that are created by the construct. It also has a property for additional ssm context to be passed (used by functions for setting up runtime-resolved env vars)

Per meeting discussion today we may need to expand this type to also provide a way to pass a "description" of the env vars to aid in typedef generation